### DISCUSSION TITLE
Packaging Compose for Desktop fails, missing ManagmentFactory -> used for SystemClock.elapsedRealtime()

### DISCUSSION THREAD

**Discussion by AlbRoehm** [GitLiveApp/firebase-kotlin-sdk#466] [Q&A] [answered]
Hey i'm struggling a bit with the following issue. When i try to run `runDistributable` (or package the desktop app) on a Compose for Desktop project (on Win and Mac the same same) i always end up with the following error. Just running `runRelease` works fine so i guess some linking/packaging is amiss where. Any help appreciated.
```
> Task :AppDesktop:runDistributable
Exception in thread "main" java.lang.NoClassDefFoundError: java/lang/management/ManagementFactory
	at android.os.SystemClock.elapsedRealtime(SystemClock.java:7)
	at com.google.firebase.StartupTime.now(StartupTime.java:54)
	at com.google.firebase.provider.FirebaseInitProvider.<clinit>(FirebaseInitProvider.java:38)
	at com.google.firebase.FirebaseApp.<init>(FirebaseApp.java:412)
	at com.google.firebase.FirebaseApp.initializeApp(FirebaseApp.java:296)
	at com.google.firebase.FirebaseApp.initializeApp(FirebaseApp.java:264)
	at dev.gitlive.firebase.FirebaseKt.initialize(firebase.kt:30)
	...
```

**Comment by AlbRoehm** [comment, marked as answer]
In case someone else looks for this. Add the `includeAllModules = true` flag fixes this issue. 
```kotlin
compose.desktop {
    application {
        mainClass = "com.example.desktop.MainKt"
        javaHome = System.getenv("JAVA_HOME")

        nativeDistributions {
            targetFormats(TargetFormat.Dmg, TargetFormat.Msi, TargetFormat.Deb)
            modules("java.management", "java.sql", "jdk.unsupported") // or includeAllModules = true  // <--
            packageName = "ExampleDesktop"
            packageVersion = "1.0.0"
        }
    }
}

```

**Comment by nbransby** [comment]
Docs for that: https://github.com/JetBrains/compose-multiplatform/blob/master/tutorials/Native_distributions_and_local_execution/README.md#configuring-included-jdk-modules

You should be able to just add the  Java Management Extensions (JMX) module

**Comment by AlbRoehm** [comment]
`modules("java.management", "java.sql", "jdk.unsupported")` seems to needed in my case to make it work. `gradle suggestRuntimeModules` can be used to figure out which ones are needed, like the docs state

**Reply by SerggioZS** [reply]
I think `suggestRuntimeModules` is the best approach. Just add the modules the runtime needs to run properlly. 
`includeAllModules = true` will increase the size of the app by a lot, 170 MB extra in my case.
